api/src/main/java/org/apache/iceberg/metrics/MetricsReport.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+public interface MetricsReport {}
```

Comment:
I think this should also expose the metrics from the underlying report. That way we only need parsers to serialize the scan- or commit-specific fields and can then delegate to a metrics parser. In addition, we can also still handle metrics from an unknown report type (minor).

Reply:
this might make sense, but I would probably defer this until we have a better picture of how commit reports look like